src/app/calendar/parcelCalendarFunctions.ts

```diff
@@ -0,0 +1,46 @@
+import supabase, { Schema } from "@/supabase";
+import { CalendarEvent } from "@/components/Calendar/Calendar";
+
+export type parcelWithClientName = Schema["parcels"] & { clients: { full_name: string } | null };
+
+export interface LocationColorMap {
+    [location: string]: { color: string; text: string };
+}
+
+const COLLECTION_DURATION_MS = 30 * 60 * 1000;
+
+export const getParcelsWithCollectionDate = async (): Promise<parcelWithClientName[]> => {
+    const response = await supabase
+        .from("parcels")
+        .select("*, clients ( full_name )")
```

Comment:
Can you explain what this does?

Reply:
It replaces the original ClientMap function

In essence, the client details in the parcel table is stored as the foreign key `client_id`, without the name of the client
However, the calendar needs to display the client name, so the join helps me link the `client_id` to the corresponding client to get their respective `full_name`